test/e2e/azure_failuredomains.go

```diff
@@ -71,13 +71,12 @@ func AzureFailureDomainsSpec(ctx context.Context, inputGetter func() AzureFailur
 			if err != nil {
 				LogWarning(err.Error())
 			}
-			return err
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(len(input.Cluster.Status.FailureDomains)).To(Equal(len(zones)))
+			for _, z := range zones {
+				g.Expect(input.Cluster.Status.FailureDomains[z]).NotTo(BeNil())
+			}
```

Comment:
It looks like we (CAPZ) populate this data during every AzureCluster reconcile, but because it is based on Azure regional zonal capabilities it should never change in practice. Then what CAPI does (according to my R&D) is simply consumes what the provider infra says (in this case AzureCluster).

So it's theoretically possible that the Cluster resource, which is what we're evaluating here, could be fetched by E2E prior to the capz controller doing its thing to populate it with FailureDomains data, but maybe that's not even possible given the serialization of this flow not even being entered into until other downstream outcomes have already occured (control plane online, nodes ready, etc).

cc @CecileRobertMichon